utilities/shared_disk.py

```
@@ -97,48 +104,110 @@ def _write_marker(ssh_conn: VMSSHConnection, file_path: str, content: str, vm_la
+        raise ValueError(f"VM '{vm_name}' in '{target_namespace}' has no running VMI")
+
+    vmi = cnv_vm.vmi
+    try:
+        for sample in TimeoutSampler(wait_timeout=300, sleep=5, func=lambda: vmi.instance):
+            if getattr(sample.status, "volumeStatus", None):
+                break
+    except TimeoutExpiredError:
+        raise ValueError(
+            f"VM '{vm_name}' in '{target_namespace}' VMI has no volumeStatus after 300s "
+            f"(phase: {getattr(vmi.instance.status, 'phase', 'unknown')})"
+        ) from None
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)